MoltenVK/MoltenVK/Commands/MVKCommandBuffer.mm

```diff
@@ -452,10 +452,11 @@
 
 const MVKMTLBufferAllocation* MVKCommandEncoder::getTempMTLBuffer(NSUInteger length) {
     const MVKMTLBufferAllocation* mtlBuffAlloc = getCommandEncodingPool()->acquireMTLBufferAllocation(length);
+	MVKMTLBufferAllocationPool* pool = mtlBuffAlloc->getPool();
 
     // Return the MTLBuffer allocation to the pool once the command buffer is done with it
     [_mtlCmdBuffer addCompletedHandler: ^(id<MTLCommandBuffer> mcb) {
-        ((MVKMTLBufferAllocation*)mtlBuffAlloc)->returnToPool();
+        pool->returnObjectSafely((MVKMTLBufferAllocation*)mtlBuffAlloc);
```

Comment:
Strange. I wonder why this matters to Thread Santizer? 

Is it thinking `_pool` could somehow change before the completion handler fires? Sigh.

Reply:
Pretty much. It doesn't know that the completion handler can't fire at all until the command buffer is committed.

Reply:
Alternatively, it thinks that the completion handler could fire *before* the object is finished being created. Yeah, I know, that should *never* happen. But its idea of "race condition" is "two or more accesses by different threads in close succession, unserialized by a mutex."